File: frontend/src/types/stock.ts
export type StockStatus = 'normal' | 'low' | 'out_of_stock';

export interface StockItem {
    product_id: string;
    product_name: string;
    sku?: string;
    quantity: number;
    threshold?: number;
    status: StockStatus;
}

export interface StockUpdate {
    quantity: number;
}

export interface StockSyncResult {
    success: boolean;
    updated_count: number;
    failed_updates: Array<{
        product_id: string;
        error: string;
    }>;
}

export interface StockUpdateResponse {
    message: string;
    product_id: string;
    quantity: number;
} 

File: backend/src/schemas/stock.py
from typing import Optional

from pydantic import BaseModel


class StockBase(BaseModel):
    product_id: str
    quantity: int
    
class StockCheck(StockBase):
    product_name: str
    sku: Optional[str] = None
    status: str  # "normal", "low", "out_of_stock"
    threshold: Optional[int] = None
    
    class Config:
        orm_mode = True
        
class StockUpdate(BaseModel):
    quantity: int
    
    class Config:
        orm_mode = True 